cpp/include/cudf/copying.hpp

```diff
@@ -498,6 +498,64 @@ std::vector<contiguous_split_result> contiguous_split(cudf::table_view const& in
  */
 std::unique_ptr<column> copy_if_else(column_view const& lhs, column_view const& rhs, column_view const& boolean_mask,
                                     rmm::mr::device_memory_resource *mr = rmm::mr::get_default_resource());
-                                 
+
+/**
+ * @brief   Returns a new column, where each element is selected from either @p lhs or 
+ *          @p rhs based on the value of the corresponding element in @p boolean_mask
+ *
+ * Selects each element i in the output column from either @p rhs or @p lhs using the following rule:
+ *          output[i] = (boolean_mask[i]) ? lhs : rhs[i]
+ *         
+ * @throws cudf::logic_error if lhs and rhs are not of the same type 
+ * @throws cudf::logic_error if boolean mask is not of type bool8
+ * @throws cudf::logic_error if boolean mask is not of the same length as rhs  
```

Comment:
Add `@throws cudf::logic_error if boolean mask has null values. ` for all copy_if_else documentation.